docs/DeveloperGuide.md

```diff
@@ -137,79 +137,85 @@ This section describes some noteworthy details on how certain features are imple
+in the `previousAppointmentLists`, and the currentAppointmentBookStatePointer still points to the current appointment book state.
 
-![UndoRedoState1](images/UndoRedoState1.png)
+![UndoState1](images/UndoState1.png)
 
-Step 3. The user executes `add n/David …​` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes `add appointment …​` to add a new appointment. The `add appointment` command also executes
+`previousAppointmentLists.push(new ArrayList<>(appointments.asUnmodifiableObservableList()))`, causing a copy of the
+current appointment book state to be saved into the `previousAppointmentLists`.
 
-![UndoRedoState2](images/UndoRedoState2.png)
+![UndoState2](images/UndoState2.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will 
+not executes `previousAppointmentLists.push(new ArrayList<>(appointments.asUnmodifiableObservableList()))`, so the 
+appointment book state will not be saved into the `previousAppointmentLists`.
 
 </div>
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Step 4. The user now decides that adding the appointment was a mistake, and decides to undo that action by 
+executing the `undo` command. The `undo` command will call `Model#undoAppointmentBook()`, which will shift
+the `currentAppointmentBookStatePointer` to the most recently saved state, pointing it to the previous appointment 
+book state, and restores the appointment book to that state.
 
-![UndoRedoState3](images/UndoRedoState3.png)
+![UndoState3](images/UndoState3.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index 0, pointing to the initial AddressBook state, then there are no previous AddressBook states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `previousAppointmentLists` is empty,
+then there are no previous AppointmentBook states to restore. The `undo` command uses `commandHistory.empty()` 
+to check if this is the case. If so, it will return an error to the user rather
```

Comment:
Should this be combined into one line? 120 characters per line does not apply to DG